### PR TITLE
Add NodeJS version

### PR DESCRIPTION
**Pull request by gaelmotte**
Here is a simple NodeJS implementation.

### CODE REVIEW

**Comment by gaelmotte** [comment]
Hum, you might want to change the MAX_STATION constant and try then benchmark `ruby bench.rb "node ."`, but you would waste your time ;)

```
Loading top stations
. done

Starting the application
. done

Loading the data
. done

Starting the benchmark
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```

File: readme.md [hunk 1]
```diff
@@ -135,6 +135,15 @@ Build: ```cargo build --release```
 
 Run the test with ```ruby test.rb ./target/release/csa_rs```
 
+
+## NodeJS implementation
+Requiers NodeJS installed
```
Comment (espadrine) [inline readme.md]: Requires Node.js installed.

**Comment by gaelmotte** [comment]
Pull request now integrates tests and JS implementation that cover functional limitation indicated in original readme

File: readme.md (hunk 1 above)
Comment (gaelmotte) [inline readme.md]: Fixed

**Comment by espadrine** [comment]
Closing this, as we have https://github.com/trainline-eu/csa-challenge/blob/master/csa.js.

**Pull request closed without merging**